Whisper.net/WhisperProcessor.cs

```diff
@@ -92,23 +93,80 @@ public unsafe (string? language, float probability) DetectLanguageWithProbabilit
             }
         }
 
-        public unsafe WhisperProcessResult Process(float[] samples)
+        public void Process(Stream waveStream)
+        {
+            var waveParser = new WaveParser(waveStream);
+
+            var samples = waveParser.GetAvgSamples();
+
+            Process(samples);
+        }
+
+        public unsafe void Process(float[] samples)
         {
-            segmentIndex = 0;
             fixed (float* pData = samples)
             {
                 NativeMethods.whisper_full(currentWhisperContext, whisperParams, (IntPtr)pData, samples.Length);
-                return new WhisperProcessResult();
             }
         }
 
-        public WhisperProcessResult Process(Stream waveStream)
+        public IAsyncEnumerable<OnSegmentEventArgs> ProcessAsync(Stream waveStream, CancellationToken cancellationToken)
         {
             var waveParser = new WaveParser(waveStream);
-
             var samples = waveParser.GetAvgSamples();

```

Comment:
Now that this method is Async, we should create GetAvgSamplesAsync which will consume the stream with ReadAsync methods.